"""Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT."""

from __future__ import annotations
import dataclasses
import requests as requests_http
from ..shared import fundingaccountresponsev2 as shared_fundingaccountresponsev2
from ..shared import inline_response_401 as shared_inline_response_401
from ..shared import inline_response_403 as shared_inline_response_403
from ..shared import inline_response_404 as shared_inline_response_404
from typing import Optional


@dataclasses.dataclass
class GetFundingAccountV2Request:
    
    funding_account_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'fundingAccountId', 'style': 'simple', 'explode': False }})  
    sensitive: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'sensitive', 'style': 'form', 'explode': True }})  
    

@dataclasses.dataclass
class GetFundingAccountV2Response:
    
    content_type: str = dataclasses.field()  
    status_code: int = dataclasses.field()  
    funding_account_response_v2: Optional[shared_fundingaccountresponsev2.FundingAccountResponseV2] = dataclasses.field(default=None)
    r"""Funding Account Response"""  
    inline_response_401: Optional[shared_inline_response_401.InlineResponse401] = dataclasses.field(default=None)
    r"""Invalid access token. May be expired or invalid"""  
    inline_response_403: Optional[shared_inline_response_403.InlineResponse403] = dataclasses.field(default=None)
    r"""The authentication does not have permissions to access the resource
    This usually occurs when there is a valid authentication instance (client or user) but they do not have the required permissions
    """  
    inline_response_404: Optional[shared_inline_response_404.InlineResponse404] = dataclasses.field(default=None)
    r"""The resource was not found or is no longer available"""  
    raw_response: Optional[requests_http.Response] = dataclasses.field(default=None)  
    